ts/deploy.ts

```diff
@@ -0,0 +1,253 @@
+import { RollupUtils } from "../types/ethers-contracts/RollupUtils";
+import { NameRegistryFactory } from "../types/ethers-contracts/NameRegistryFactory";
+import { GovernanceFactory } from "../types/ethers-contracts/GovernanceFactory";
+import { MerkleTreeUtilsFactory } from "../types/ethers-contracts/MerkleTreeUtilsFactory";
+import { MerkleTreeUtils } from "../types/ethers-contracts/MerkleTreeUtils";
+import { LoggerFactory } from "../types/ethers-contracts/LoggerFactory";
+import { TokenRegistryFactory } from "../types/ethers-contracts/TokenRegistryFactory";
+import { PobFactory } from "../types/ethers-contracts/PobFactory";
+import { CreateAccountFactory } from "../types/ethers-contracts/CreateAccountFactory";
+import { AirdropFactory } from "../types/ethers-contracts/AirdropFactory";
+import { TransferFactory } from "../types/ethers-contracts/TransferFactory";
+import { BurnConsentFactory } from "../types/ethers-contracts/BurnConsentFactory";
+import { BurnExecutionFactory } from "../types/ethers-contracts/BurnExecutionFactory";
+import { TestTokenFactory } from "../types/ethers-contracts/TestTokenFactory";
+import { DepositManagerFactory } from "../types/ethers-contracts/DepositManagerFactory";
+import { IncrementalTreeFactory } from "../types/ethers-contracts/IncrementalTreeFactory";
+import { RollupFactory } from "../types/ethers-contracts/RollupFactory";
+import { RollupRedditFactory } from "../types/ethers-contracts/RollupRedditFactory";
+
+import { ethers, Signer, Contract } from "ethers";
+
+interface DeploymentParameters {
+    MAX_DEPTH: number;
+    MAX_DEPOSIT_SUBTREE_DEPTH: number;
+}
+
+async function deployAll(
+    signer: Signer,
+    parameters: DeploymentParameters
+): Promise<{ [key: string]: Contract }> {
+    // deploy libs
+    const ecVerify = await new EcVerifyFactory(signer).deploy();
+    await ecVerify.deployed();
+
+    const paramManager = await new ParamManagerFactory(signer).deploy();
+    await paramManager.deployed();
+
+    const rollupUtils = await new RollupUtilsFactory(signer).deploy();
+    await rollupUtils.deployed();
+
+    // deploy name registry
+    const nameRegistry = await new NameRegistryFactory(signer).deploy();
+    await nameRegistry.deployed();
+
+    // deploy governance
+    const governance = await new GovernanceFactory(signer).deploy(
+        parameters.MAX_DEPTH,
+        parameters.MAX_DEPOSIT_SUBTREE_DEPTH
+    );
+    await governance.deployed();
+    await nameRegistry.registerName(
+        await paramManager.Governance(),
+        governance.address
+    );
+
+    // deploy logger
+    const logger = await new LoggerFactory(signer).deploy();
+    await logger.deployed();
+    await nameRegistry.registerName(
+        await paramManager.LOGGER(),
+        logger.address
+    );
+
+    const allLinkRefs = {
+        __$b941c30c0f5422d8b714f571f17d94a5fd$__: paramManager.address,
+        __$a6b8846b3184b62d6aec39d1f36e30dab3$__: rollupUtils.address
```

Comment:
linking is annoying. But the factory from the typechain/ethers-v4 requires this exact inputs of linking reference.

These are produced by `solidityHash(filepath+libraryname)`

Reply:
Is there no workaround for this? :((.